src/H5VLcallback.c

```diff
@@ -1039,6 +1681,37 @@ H5VL_attr_create(const H5VL_object_t *vol_obj, const H5VL_loc_params_t *loc_para
     FUNC_LEAVE_NOAPI(ret_value)
 } /* end H5VL_attr_create() */
 
+/*-------------------------------------------------------------------------
+ * Function:    H5VL__attr_create_api_op
+ *
+ * Purpose:     Callback for common API wrapper to create an attribute
+ *
+ * Return:      Success:    Non-negative
+ *              Failure:    Negative
+ *
+ *-------------------------------------------------------------------------
+ */
+static herr_t
+H5VL__attr_create_api_op(void *_ctx)
+{
+    H5VL_attr_create_ctx_t *ctx       = (H5VL_attr_create_ctx_t *)_ctx; /* Get pointer to context */
+    herr_t                  ret_value = SUCCEED;                        /* Return value */
+
+    FUNC_ENTER_STATIC
+
+    /* Sanity check */
+    HDassert(ctx);
+
+    /* Call the corresponding internal VOL routine */
+    if (NULL == (ctx->ret_value =
+                     H5VL__attr_create(ctx->obj, ctx->loc_params, ctx->cls, ctx->name, ctx->type_id,
+                                       ctx->space_id, ctx->acpl_id, ctx->aapl_id, ctx->dxpl_id, ctx->req)))
+        HGOTO_ERROR(H5E_VOL, H5E_CANTCREATE, FAIL, "unable to create attribute")
+
+done:
+    FUNC_LEAVE_NOAPI(ret_value)
+} /* end H5VL__attr_create_api_op() */
```

Comment:
It is a lot of code and repetitions for something that should just be inline / one line.